sign_api/src/rpc/params/shared_types.rs

```diff
@@ -0,0 +1,132 @@
+use anyhow::{Ok, Result};
+use serde::{Deserialize, Serialize};
+
+#[derive(Debug, Serialize, PartialEq, Eq, Deserialize, Clone, Default)]
+#[serde(rename_all = "camelCase")]
+pub struct Metadata {
+    pub description: String,
+    pub url: String,
+    pub icons: Vec<String>,
+    pub name: String,
+}
+
+#[derive(Debug, Serialize, PartialEq, Eq, Deserialize, Clone, Default)]
+pub struct Relay {
+    pub protocol: String,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    #[serde(default)]
+    pub data: Option<String>,
+}
+
+#[derive(Debug, Serialize, PartialEq, Eq, Deserialize, Clone, Default)]
+#[serde(rename_all = "camelCase")]
+pub struct Namespace {
+    pub chains: Vec<String>,
+    pub methods: Vec<String>,
+    pub events: Vec<String>,
```

Comment:
Might be cleaner to use `HashSet`, checking whether `Namespace` implements all required elements, potentially becomes a single function call.

Reply:
Done.